projects/plugins/jetpack/_inc/client/traffic/seo.jsx

```diff
@@ -50,26 +50,16 @@ class SeoComponent extends React.Component {
 						) }
 					</span>
 				</SettingsGroup>
-				{ ! this.props.isUnavailableInOfflineMode( 'seo-tools' ) &&
-					includes(
-						[
-							'is-premium-plan',
-							'is-business-plan',
-							'is-daily-security-plan',
-							'is-realtime-security-plan',
-							'is-complete-plan',
-						],
-						planClass
-					) && (
-						<Card
-							compact
-							className="jp-settings-card__configure-link"
-							onClick={ this.trackConfigureClick }
-							href={ this.props.configureUrl }
-						>
-							{ __( 'Customize your SEO settings', 'jetpack' ) }
-						</Card>
-					) }
+				{ ! this.props.isUnavailableInOfflineMode( 'seo-tools' ) && (
+					<Card
+						compact
+						className="jp-settings-card__configure-link"
+						onClick={ this.trackConfigureClick }
+						href={ this.props.configureUrl }
+					>
+						{ __( 'Customize your SEO settings', 'jetpack' ) }
+					</Card>
+				) }
```

Comment:
Could we allow one to activate the feature here, if it's not already active, and only show the link to customize once the module is active? 
This is what we do for the Sharing or Publicize modules for example, or on the "My Plan" page for the SEO tools.

Reply:
That would be a nice change, though it seems like the ideal goal would be to have the meta fields available in wp-admin, which would also work well if we want this to work for a user-less connection. I'll need to explore this a bit more to see what implementing the controls to wp-admin would look like.

Reply:
Sure thing. I think adding those fields to wp-admin is a bit beyond the scope of this PR though. I would personally recommend that we focus on the move to free in this PR, and then develop the new interface to manage settings in wp-admin in a follow-up PR.

Reply:
Tracking the WP Admin portion in: https://github.com/Automattic/jetpack/pull/18810

Reply:
I still think we should allow one to activate the module from here now. In the follow-up PR we can expand on things and have the additional settings show up instead of just the link to Calypso, but until then, it would be nice if one were able to activate the module here when it's inactive, instead of having to go to the My Plan page (where I wouldn't think to go when I have a free plan, so I think we could even remove the card from there at this point).

This would also allow us to get rid of the logic with `getModuleOverridenBanner`, since the module would use a more traditional module toggle.

Something like this?

```diff
diff --git a/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx b/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx
index 7964e86de..a64ac52d0 100644
--- a/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx
+++ b/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx
@@ -15,7 +15,6 @@ import Button from 'components/button';
 import {
 	FEATURE_SECURITY_SCANNING_JETPACK,
 	FEATURE_SITE_BACKUPS_JETPACK,
-	FEATURE_SEO_TOOLS_JETPACK,
 	FEATURE_VIDEO_HOSTING_JETPACK,
 	FEATURE_GOOGLE_ANALYTICS_JETPACK,
 	FEATURE_WORDADS_JETPACK,
@@ -299,8 +298,6 @@ export const SettingsCard = props => {
 				return 'inactive' === props.getModuleOverride( 'wordads' );
 			case FEATURE_GOOGLE_ANALYTICS_JETPACK:
 				return 'inactive' === props.getModuleOverride( 'google-analytics' );
-			case FEATURE_SEO_TOOLS_JETPACK:
-				return 'inactive' === props.getModuleOverride( 'seo-tools' );
 			case FEATURE_SEARCH_JETPACK:
 				return 'inactive' === props.getModuleOverride( 'search' );
 			default:
@@ -308,22 +305,18 @@ export const SettingsCard = props => {
 		}
 	};
 
-	// We only want to show this banner for Google Analytics and SEO Tools because
+	// We only want to show this banner for Google Analytics because
 	// they don't use the ModuleToggle for their UI.
-	const getModuleOverridenBanner = () => {
+	const getGoogleAnalyticsOverridenBanner = () => {
 		if ( ! featureIsOverriden() ) {
 			return false;
 		}
-		switch ( feature ) {
-			case FEATURE_GOOGLE_ANALYTICS_JETPACK:
-				const googleAnalytics = props.getModule( 'google-analytics' );
-				return <ModuleOverridenBanner moduleName={ googleAnalytics.name } />;
-			case FEATURE_SEO_TOOLS_JETPACK:
-				const seoTools = props.getModule( 'seo-tools' );
-				return <ModuleOverridenBanner moduleName={ seoTools.name } />;
-			default:
-				return null;
+		if ( feature !== FEATURE_GOOGLE_ANALYTICS_JETPACK ) {
+			return null;
 		}
+
+		const googleAnalytics = props.getModule( 'google-analytics' );
+		return <ModuleOverridenBanner moduleName={ googleAnalytics.name } />;
 	};
 
 	const children = showChildren() && props.children;
@@ -342,7 +335,7 @@ export const SettingsCard = props => {
 	}
 
 	return (
-		getModuleOverridenBanner() || (
+		getGoogleAnalyticsOverridenBanner() || (
 			<form
 				{ ...( moduleId ? { id: moduleId } : null ) }
 				className={ `jp-form-settings-card` }
@@ -398,7 +391,6 @@ export default connect( state => {
 		securityProUpgradeUrl: getUpgradeUrl( state, 'settings-security-pro' ),
 		securityPremiumUpgradeUrl: getUpgradeUrl( state, 'settings-security-premium' ),
 		gaUpgradeUrl: getUpgradeUrl( state, 'settings-ga' ),
-		seoUpgradeUrl: getUpgradeUrl( state, 'settings-seo' ),
 		searchUpgradeUrl: getUpgradeUrl( state, 'jetpack-search' ),
 		spamUpgradeUrl: getUpgradeUrl( state, 'settings-spam' ),
 		multisite: isMultisite( state ),
diff --git a/projects/plugins/jetpack/_inc/client/lib/plans/constants.js b/projects/plugins/jetpack/_inc/client/lib/plans/constants.js
index dddae6dbe..e151ae0b1 100644
--- a/projects/plugins/jetpack/_inc/client/lib/plans/constants.js
+++ b/projects/plugins/jetpack/_inc/client/lib/plans/constants.js
@@ -154,7 +154,6 @@ export const FEATURE_REVENUE_GENERATION_JETPACK = 'revenue-generation-jetpack';
 export const FEATURE_VIDEO_HOSTING_JETPACK = 'video-hosting-jetpack';
 export const FEATURE_SECURITY_ESSENTIALS_JETPACK = 'security-essentials-jetpack';
 export const FEATURE_PRIORITY_SUPPORT_JETPACK = 'priority-support-jetpack';
-export const FEATURE_SEO_TOOLS_JETPACK = 'seo-tools-jetpack';
 export const FEATURE_WORDADS_JETPACK = 'wordads-jetpack';
 export const FEATURE_GOOGLE_ANALYTICS_JETPACK = 'google-analytics-jetpack';
 export const FEATURE_SEARCH_JETPACK = 'search-jetpack';
diff --git a/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx b/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
index 54ca777e1..19cd23e5d 100644
--- a/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
+++ b/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
@@ -68,56 +68,11 @@ class MyPlanBody extends React.Component {
 		this.trackPlansClick( 'activate_videopress' );
 	};
 
-	activateSeo = () => {
-		this.props.activateModule( 'seo-tools' );
-		this.trackPlansClick( 'activate_seo' );
-	};
-
 	activateGoogleAnalytics = () => {
 		this.props.activateModule( 'google-analytics' );
 		this.trackPlansClick( 'activate_ga' );
 	};
 
-	getSeoToolsCard = () => {
-		return (
-			<div className="jp-landing__plan-features-card">
-				<div className="jp-landing__plan-features-img">
-					<img
-						src={ imagePath + '/jetpack-performance-icon.svg' }
-						className="jp-landing__plan-features-icon"
-						alt={ __( 'Site stats showing an evolution in traffic and engagement', 'jetpack' ) }
-					/>
-				</div>
-				<div className="jp-landing__plan-features-text">
-					<h3 className="jp-landing__plan-features-title">{ __( 'SEO Tools', 'jetpack' ) }</h3>
-					<p>
-						{ __(
-							'Advanced SEO tools to help your site get found when people search for relevant content.',
-							'jetpack'
-						) }
-					</p>
-					{ this.props.isModuleActivated( 'seo-tools' ) ? (
-						<Button
-							onClick={ this.handleButtonClickForTracking( 'configure_seo' ) }
-							href={ getRedirectUrl( 'calypso-marketing-traffic', {
-								site: this.props.siteRawUrl,
-							} ) }
-						>
-							{ __( 'Configure site SEO', 'jetpack' ) }
-						</Button>
-					) : (
-						<Button
-							onClick={ this.activateSeo }
-							disabled={ this.props.isActivatingModule( 'seo-tools' ) }
-						>
-							{ __( 'Activate SEO tools', 'jetpack' ) }
-						</Button>
-					) }
-				</div>
-			</div>
-		);
-	};
-
 	render() {
 		let planCard = '';
 		const planClass = 'offline' !== this.props.plan ? getPlanClass( this.props.plan ) : 'offline';
@@ -502,8 +457,6 @@ class MyPlanBody extends React.Component {
 							</div>
 						) }
 
-						{ 'inactive' !== this.props.getModuleOverride( 'seo-tools' ) && this.getSeoToolsCard() }
-
 						{ isPlanPremiumOrBetter &&
 							'inactive' !== this.props.getModuleOverride( 'google-analytics' ) && (
 								<div className="jp-landing__plan-features-card">
@@ -757,8 +710,6 @@ class MyPlanBody extends React.Component {
 							) }
 						</div>
 
-						{ 'inactive' !== this.props.getModuleOverride( 'seo-tools' ) && this.getSeoToolsCard() }
-
 						<div className="jp-landing__plan-features-card">
 							<div className="jp-landing__plan-features-img">
 								<img
diff --git a/projects/plugins/jetpack/_inc/client/traffic/seo.jsx b/projects/plugins/jetpack/_inc/client/traffic/seo.jsx
index cacfb2a09..a5e78d050 100644
--- a/projects/plugins/jetpack/_inc/client/traffic/seo.jsx
+++ b/projects/plugins/jetpack/_inc/client/traffic/seo.jsx
@@ -1,8 +1,7 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { connect } from 'react-redux';
+import React, { Component } from 'react';
 import { __, _x } from '@wordpress/i18n';
 
 /**
@@ -11,60 +10,66 @@ import { __, _x } from '@wordpress/i18n';
 import analytics from 'lib/analytics';
 import Card from 'components/card';
 import getRedirectUrl from 'lib/jp-redirect';
-import { FEATURE_SEO_TOOLS_JETPACK } from 'lib/plans/constants';
+import { ModuleToggle } from 'components/module-toggle';
 import { withModuleSettingsFormHelpers } from 'components/module-settings/with-module-settings-form-helpers';
 import SettingsCard from 'components/settings-card';
 import SettingsGroup from 'components/settings-group';
-import { getSitePlan } from 'state/site';
 
-class SeoComponent extends React.Component {
-	trackConfigureClick = () => {
-		analytics.tracks.recordJetpackClick( 'configure-seo' );
-	};
+export const SEO = withModuleSettingsFormHelpers(
+	class extends Component {
+		trackConfigureClick = () => {
+			analytics.tracks.recordJetpackClick( 'configure-seo' );
+		};
 
-	render() {
-		return (
-			<SettingsCard
-				{ ...this.props }
-				header={ _x( 'Search engine optimization', 'Settings header', 'jetpack' ) }
-				feature={ FEATURE_SEO_TOOLS_JETPACK }
-				hideButton
-			>
-				<SettingsGroup
-					disableInOfflineMode
-					module={ { module: 'seo-tools' } }
-					support={ {
-						text: __(
-							'Allows you to optimize your site and its content for better results in search engines.',
-							'jetpack'
-						),
-						link: getRedirectUrl( 'jetpack-support-seo-tools' ),
-					} }
+		render() {
+			const isActive = this.props.getOptionValue( 'seo-tools' );
+
+			return (
+				<SettingsCard
+					{ ...this.props }
+					header={ _x( 'Search engine optimization', 'Settings header', 'jetpack' ) }
+					feature={ 'seo-tools-jetpack' }
+					module="seo-tools"
+					hideButton
 				>
-					<span>
-						{ __(
-							'Take control of the way search engines represent your site. With Jetpack’s SEO tools you can preview how your content will look on popular search engines and change items like your site name and tagline in seconds.',
-							'jetpack'
-						) }
-					</span>
-				</SettingsGroup>
-				{ ! this.props.isUnavailableInOfflineMode( 'seo-tools' ) && (
-					<Card
-						compact
-						className="jp-settings-card__configure-link"
-						onClick={ this.trackConfigureClick }
-						href={ this.props.configureUrl }
+					<SettingsGroup
+						disableInOfflineMode
+						module={ { module: 'seo-tools' } }
+						support={ {
+							text: __(
+								'Allows you to optimize your site and its content for better results in search engines.',
+								'jetpack'
+							),
+							link: getRedirectUrl( 'jetpack-support-seo-tools' ),
+						} }
 					>
-						{ __( 'Customize your SEO settings', 'jetpack' ) }
-					</Card>
-				) }
-			</SettingsCard>
-		);
+						<p>
+							{ __(
+								'Take control of the way search engines represent your site. With Jetpack’s SEO tools you can preview how your content will look on popular search engines and change items like your site name and tagline in seconds.',
+								'jetpack'
+							) }
+						</p>
+						<ModuleToggle
+							slug="seo-tools"
+							activated={ isActive }
+							toggling={ this.props.isSavingAnyOption( 'seo-tools' ) }
+							toggleModule={ this.props.toggleModuleNow }
+						>
+							{ __( 'Customize your SEO settings', 'jetpack' ) }
+						</ModuleToggle>
+					</SettingsGroup>
+					{ isActive && ! this.props.isOfflineMode && (
+						<Card
+							compact
+							className="jp-settings-card__configure-link"
+							onClick={ this.trackConfigureClick }
+							href={ this.props.configureUrl }
+						>
+							{ __( 'Customize your SEO settings', 'jetpack' ) }
+						</Card>
+					) }
+				</SettingsCard>
+			);
+		}
 	}
-}
-
-export const SEO = connect( state => {
-	return {
-		sitePlan: getSitePlan( state ),
-	};
-} )( withModuleSettingsFormHelpers( SeoComponent ) );
+);

```

Reply:
Awesome! Looks good and test well on my end 👍 

* https://github.com/Automattic/jetpack/pull/17309/commits/2631a3c950fbe2c91d11d125f9ab273c8c196051
* https://github.com/Automattic/jetpack/pull/17309/commits/f0ec8fdb2ff20cc94b424cb43a1f8af9edf8eebb
* https://github.com/Automattic/jetpack/pull/17309/commits/a7c42c6c4ee9eaa88a982fd8cc0c3dc442a47125